src/formats/tiff.rs

```diff
@@ -55,7 +103,16 @@ pub fn size<R: BufRead + Seek>(reader: &mut R) -> ImageResult<ImageSize> {
             // RATIONAL | SRATIONAL
             5 | 10 => 4 * 2,
             // DOUBLE | LONG8 | SLONG8 | IFD8
-            12 | 16 | 17 | 18 => 8,
+            12 | 16 | 17 | 18 => {
+                if tiff_type == Type::Tiff {
+                    return Err(std::io::Error::new(
+                        std::io::ErrorKind::InvalidData,
+                        "Invalid IFD type for standard TIFF",
+                    )
+                    .into());
+                }
+                8
+            }
```

Comment:
Actually , maybe DOUBLE is supported by regular TIFF.
I need to investigate this more.

Reply:
fixed